### DISCUSSION TITLE
Grouping users based on respective Admin Account

### DISCUSSION THREAD

**Discussion by rashmimaharana** [drakkan/sftpgo#734] [Ideas]
Hi @drakkan 

We would like to group users based on the admin account who creates them. So that admin1 can't see the users created by admin2 and vice versa. Do we have this user grouping feature.
Thanks for all your response.

Thanks,
Rashmi Maharana

**Comment by drakkan** [comment]
Hi,

a group feature is planned but someone has to sponsor it. I don't have a definite schedule sorry

**Reply by rashmimaharana** [reply]
Thanks @drakkan 

Will you welcome if we contribute by doing the code changes required for this feature and will raise PR for same ?

Thanks,
Rashmi Maharana

**Reply by drakkan** [reply]
Yes, but it must be done in an extensible way, a simple foreign key between users and admin is not enough. We need to create a separate group/vsite table. Admins will be associated to a group and users created by a specific admin will be added to the same group. We probably need to update the jwt tokens to add the group id and probably also add a mapping field for OIDC auth.

I have not thought about the various details but the basic idea is this. I think this is a big change.

If you need this feature please work on it in a separate branch (it must be public for the SFTPGo license), I'll take a look from time to time and once you have something working and well tested please send a PR. Even if it cannot be merged as is your implementation could be a starting point. Thanks

**Reply by rashmimaharana** [reply]
Thanks @drakkan for the heads up. Sure we will do it as an extensible way. I will post you once we will cover all aspects.

**Reply by rashmimaharana** [reply]
Thanks @drakkan for the heads up. We will do it as an extensible way. I will post you the details once we are done. Will be done in separate branch.

**Comment by krushnadattaswain** [comment]
@drakkan  My Organisation will sponsor this feature, can you help on this grouping feature.

**Reply by drakkan** [reply]
> @drakkan My Organisation will sponsor this feature, can you help on this grouping feature.

Hi,

thank you for your offer, please contact me at the email address you see in each commit and we will discuss the technical details and the amount for sponsorship and maintenance

**Reply by glasserre-pdw** [reply]
> @drakkan My Organisation will sponsor this feature, can you help on this grouping feature.

Many thanks !
